src/frame.h

```diff
@@ -69,6 +71,10 @@ struct frame_additional_data : frame_header
 
     uint32_t raw_size = 0;     // The frame transmitted size (payload only)
 
+    // Used by software-device, to save metadata that doesn't have enough room in metadata_blob
+    // (saving sizeof(rs2_frame_metadata_value) + sizeof(rs2_metadata_type) for each blob value)
+    std::map< rs2_frame_metadata_value, rs2_metadata_type > sw_device_extra_data;
```

Comment:
What I meant was that we should changes `metadata_blob` to a map. Possible?

Reply:
Changing the blob would be a big change, that may impact many non-dds related locations, and tested carefully. Not as part of this PR.

Reply:
So it should be the next step -- I don't think we should add another mechanism when a single (and better) mechanism is known.

Reply:
Added to the [to-do list](https://rsjira.intel.com/browse/LRS-585)